components/blobserve/cmd/run.go

```diff
@@ -100,6 +105,31 @@ var runCmd = &cobra.Command{
 			log.WithField("addr", cfg.PrometheusAddr).Info("started Prometheus metrics server")
 		}
 
+		if cfg.ReadinessProbeAddr != "" {
+			// use the first layer as source for the tests
+			if len(cfg.BlobServe.Repos) < 1 {
+				log.Panic("the configuration file do not contain static layers")
+			}
+
+			var repository string
+			for k := range cfg.BlobServe.Repos {
+				repository = k
+			}
+			staticLayerHost := strings.Split(repository, "/")[0]
```

Comment:
This will not do the right thing if the repo is `docker.io`. It would probably still work, just refer to the wrong host. This is how we'd handle the case correctly:

```Go
repo, err := reference.ParseFamiliarName(repository)
if err != nil {
    log.WithError(err).WithField("repo", repository).Fatal("cannot parse repository reference")
}
staticLayerHost := reference.Domain(repo)
```